docs/remote-interface.md

```diff
@@ -0,0 +1,59 @@
+# Remote interface
+
+## Introduction
+
+```
```

Comment:
ASCII-based diagram should be moved to the paragraph next to high-level descriptions.